### ISSUE TITLE
Feed - Video Long Press Menu - Redesign

### ISSUE THREAD

**Issue by MFlisar** [TeamNewPipe/NewPipe#11416]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

When long pressing a video in the abo feed a menu pops up with options like:

* play in background
* play next
* add to queue
* mark as watched
* mark as unwathed
* add to playlist
* download
* share

Especially mark as watched/unwatched is popping up based on if the video was not yet watched, partly watched or nearly fully watched or fully watched.

**Suggestion:**

Either add icons to menu or better, split it up into a "header" showing actions as icons only (actions whos icons are so common that everyone knows their meaning immediately) and only show more app specific actions as text menus (but also add icons).

As the menu is quite large (and not static) the user must always waste time reading the menu items to find the correct menu item.

**Suggested Layout**

* Horizontal Header with icon actions only for (no scrolling but evenly distributed icons): Watched / Unwatched / Download / Share
* List: Rest of the items

The watched/unwatched icons should be enabled/disabled instead of added/removed based on if they are currently applicable or not.

**Side Note**

The horizontal header could be icon + text as well if the text is shortened and displayed *below* the icon. The texts Watched / Unwatched / Download / Share would do it and should be short in most languages...

### Why do you want this feature?

I use this heavily to mark videos as watched, either after watching it partly or directly when I did not watch it at all.

I always have to read the texts to find the correct menu entry because there is no visually hint that is detectable fast.

A big problem is that there are no icons and that the list is not static but dynamically build so the user has to do following whenever the menu pops up:

* scan the list and find the correct menu entry - because the menu does not always look the same
* read the menu exactly because there may not be the entry the user expects (e.g. a nearly fully watched video won't have the "mark as watched" entry anymore because it is already watched >=95% and will be removed on the next feed reload anyway).

**Comment by MFlisar** [comment]
> You use the marking feature heavily, so you're asking for those buttons. Other users may want other buttons to be shown prominently.

Correct. But the basic issue for me is no icons and "jumping" menu entries which makes it hard to detect the correct menu entry effectively and fast.

Which menu items are shown on top could even be customisable... Although this thing is not that important for me, it just fits here and would be a nice addition and would make the large menu more cleaned up.

**Comment by Isayso** [comment]
I use "send to" very often and would like to have the last target as an icon, like other apps do.
